basic/src/tasks/components/CreateTaskForm.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { createTask, getTags, useQuery } from "wasp/client/operations";
+import { Tag } from "wasp/entities";
+import { Button } from "../../shared/components/Button";
+import { Input } from "../../shared/components/Input";
+import { CreateTagDialog } from "../../tags/components/CreateTagDialog";
+import { TagLabel } from "../../tags/components/TagLabel";
+
+interface CreateTaskFormValues {
+  description: string;
+  tagIds: number[];
+}
+
+const initialState: CreateTaskFormValues = {
+  description: "",
+  tagIds: [],
+};
+
+export function CreateTaskForm() {
+  const [state, setState] = React.useState<CreateTaskFormValues>(initialState);
+  const { data: tags } = useQuery(getTags);
+
+  async function createNewTask(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    try {
+      await createTask(state);
+      setState(initialState);
+    } catch (err: unknown) {
+      window.alert(`Error while creating task: ${String(err)}`);
+    }
+  }
+
+  function toggleTag(id: Tag["id"]) {
+    if (state.tagIds.includes(id)) {
+      setState({
+        ...state,
+        tagIds: state.tagIds.filter((tagId) => tagId !== id),
+      });
+    } else {
+      setState({ ...state, tagIds: [...state.tagIds, id] });
+    }
+  }
+
+  return (
+    <form onSubmit={createNewTask} className="flex w-full flex-col gap-6">
+      <h2 className="text-xl font-semibold">Create a new task</h2>
+      <Input
+        required
+        label="Description"
+        placeholder="What do I need to do?"
+        value={state.description}
+        onChange={(e) => setState({ ...state, description: e.target.value })}
+      />
+      <div className="flex flex-col gap-2">
+        <span>Select tags</span>
+        <div className="flex flex-wrap gap-4">
+          <ul className="flex flex-wrap gap-2">
```

Comment:
```suggestion
          {tags && tags.length > 0 && (<ul className="flex flex-wrap gap-2">
```

Hide the `ul` when there are no tags to avoid broken layout.

![Screenshot 2025-06-09 at 11 40 31](https://github.com/user-attachments/assets/9f064649-4b40-4841-a9af-db667717bf87)